controllers/operator_controller_test.go

```diff
@@ -151,37 +152,234 @@ var _ = Describe("Reconcile Test", func() {
 								},
 							},
 						},
-					})
+					}
+					err := cl.Create(ctx, bd)
 					Expect(err).NotTo(HaveOccurred())
+				})
 
-					By("running reconcile")
-					res, err := reconciler.Reconcile(ctx, ctrl.Request{NamespacedName: opKey})
-					Expect(res).To(Equal(ctrl.Result{}))
-					Expect(err).NotTo(HaveOccurred())
+				When("the BundleDeployment status is stale", func() {
```

Comment:
I think within the `When("the expected BundleDeployment already exists")`, maybe we need a slightly different organization?

When the BD spec is out of date:
 - we expect to patch (and therefore we expect the status to be outdated after the patch)

When the BD spec is up-to-date:
  - we expect NOT to patch (that's the `PIt` that needs to be implemented with a wrapper or fake client)
  - we expect the various BD status possibilities to be mapped to the appropriate Operator status.

WDYT?

Reply:
Re-arranged the tests in that block for check if the spec is updated.